code/scripting/api/libs/mission.cpp

```diff
@@ -1340,6 +1419,45 @@ ADE_FUNC(createShip,
 		return ade_set_error(L, "o", l_Ship.Set(object_h()));
 }
 
+ADE_FUNC(createProp,
+	l_Mission,
+	"[string Name, propclass Class /* First prop class by default */, orientation Orientation=null, vector Position /* null vector by default */]",
+	"Creates a prop and returns a handle to it using the specified name, class, world orientation, and world position.",
+	"prop",
+	"Prop handle, or invalid prop handle if prop couldn't be created")
+{
+	const char* name = nullptr;
+	int pclass       = -1;
+	matrix_h* orient = nullptr;
+	vec3d pos        = vmd_zero_vector;
+	ade_get_args(L, "|sooo", &name, l_Propclass.Get(&pclass), l_Matrix.GetPtr(&orient), l_Vector.Get(&pos));
+
+	if (Prop_info.empty()) {
+		return ade_set_error(L, "o", l_Prop.Set(object_h()));
+	}
+
+	matrix *real_orient = &vmd_identity_matrix;
+	if(orient != nullptr)
+	{
+		real_orient = orient->GetMatrix();
+	}
+
+	if (pclass == -1) {
+		return ade_set_error(L, "o", l_Prop.Set(object_h()));
+	}
```

Comment:
The function description calls the default for the prop class "the first prop class".
To make it consistent, either this needs to then set pclass to 0 instead of -1, or you need to change the documentation to say that it just won't spawn if the  propclass is unknown